### PR TITLE
use gas to tx fee

### PR DESCRIPTION
**Pull request by flybikeGx**

### CODE REVIEW

File: vm/database/gas_handler.go [hunk 1]
```diff
@@ -12,19 +12,29 @@ const (
 	gasPledgeSuffix     = "-gp"
 )
 
+const (
+	decPledge  = 8
+	decGasRate = 8
+	decGas     = 8
```
Comment (wangyu0air) [inline vm/database/gas_handler.go]: i thought decimal of gas is '2'

File: vm/isolator.go [hunk 2]
```diff
@@ -38,13 +44,15 @@ func (e *Isolator) Prepare(bh *block.BlockHead, db *database.Visitor, logger *il
...
+		err := checkTxParams(t)
+		if err != nil {
+			return err
+		}
 	}
 	e.publisherID = t.Publisher
-	bl := e.h.DB().Balance(e.publisherID)
-	if bl < 0 || bl < t.GasPrice*t.GasLimit {
+	gas := e.h.CurrentGas(e.publisherID)
+	if gas.ToFloat() < float64(t.GasPrice*t.GasLimit)/100 {
```
Comment (wangyu0air) [inline vm/isolator.go]: don't use float64

File: test/integration/verifier_test.go [hunk 3]
```diff
@@ -16,22 +16,58 @@ import (
...
 	defer s.Clear()
-	s.Visitor.SetBalance(testID[0], 10000000)
-
 	kp := prepareAuth(t, s)
 
-	r, err := s.Call("iost.system", "Transfer", fmt.Sprintf(`["%v","%v","%v"]`, testID[0], testID[2], 0.0001), testID[0], kp)
+	s.Visitor.SetBalance(kp.ID, 100000000)
+	s.SetGas(kp.ID, 1000)
+
+	r, err := s.Call("iost.system", "Transfer", fmt.Sprintf(`["%v","%v","%v"]`, testID[0], testID[2], 0.0001), kp.ID, kp)
```
Comment (wangyu0air) [inline test/integration/verifier_test.go]: use iost.token transfer instead iost.system

File: test/integration/verifier_test.go [hunk 4]
```diff
@@ -16,22 +16,58 @@ import (
...
+
+}
+
+func TestSetCode(t *testing.T) {
+	ilog.Stop()
+	s := NewSimulator()
+	defer s.Clear()
+	kp := prepareAuth(t, s)
+	s.SetAccount(account.NewInitAccount(kp.ID, kp.ID, kp.ID))
+	s.Visitor.SetBalance(kp.ID, 100000000)
```
Comment (wangyu0air) [inline test/integration/verifier_test.go]: remove SetBalance

File: vm/database/gas_handler.go (hunk 1 above)
Comment (flybikeGx) [reply vm/database/gas_handler.go]: just a temporary value

**Pull request merged**